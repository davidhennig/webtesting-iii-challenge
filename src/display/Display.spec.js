// Test away!
import React from "react";
import Display from "./Display";
import { render } from "@testing-library/react";

test("Check if Closed", () => {
  const { getByTestId, rerender } = render(<Display closed />);
  getByTestId("isItClosed");

  rerender(<Display closed={false} />);
  getByTestId("isItClosed");
});

test("Check if Locked", () => {
  const { getByTestId, rerender } = render(<Display locked />);
  getByTestId("isItLocked");

  rerender(<Display locked={false} />);
  getByTestId("isItLocked");
});

test("Check if Green Green", () => {
  const className = "led green-led";
  const locked = false;
  const closed = false;
  const display = render(
    <Display closed={closed} locked={locked} className={className} />
  );
  expect(display.baseElement).toMatchSnapshot();
});

test("Check if Red Green", () => {
  const className = "led red-led";
  const locked = false;
  const display = render(
    <Display closed locked={locked} className={className} />
  );
  expect(display.baseElement).toMatchSnapshot();
});

test("Check if Red Red", () => {
  const className = "led red-led";
  const locked = true;
  const closed = true;
  const display = render(
    <Display closed={closed} locked={locked} className={className} />
  );
  expect(display.baseElement).toMatchSnapshot();
});
