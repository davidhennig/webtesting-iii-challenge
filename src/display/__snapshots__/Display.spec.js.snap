// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Green Green 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
        data-testid="isItLocked"
      >
        Unlocked
      </div>
      <div
        class="led green-led"
        data-testid="isItClosed"
      >
        Open
      </div>
    </div>
  </div>
</body>
`;

exports[`Check if Red Green 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led green-led"
        data-testid="isItLocked"
      >
        Unlocked
      </div>
      <div
        class="led red-led"
        data-testid="isItClosed"
      >
        Closed
      </div>
    </div>
  </div>
</body>
`;

exports[`Check if Red Red 1`] = `
<body>
  <div>
    <div
      class="display panel"
    >
      <div
        class="led red-led"
        data-testid="isItLocked"
      >
        Locked
      </div>
      <div
        class="led red-led"
        data-testid="isItClosed"
      >
        Closed
      </div>
    </div>
  </div>
</body>
`;
